### PR TITLE
Extract BinderAnchor and ResultAnchor from LambdaContainer

### PR DESCRIPTION
**Pull request by Flumble**
Required for case blocks

### CODE REVIEW

**Comment by wandernauta** [comment]
Thanks, I'll take a look.

File: Code/src/main/java/nl/utwente/viskell/ui/components/BinderAnchor.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import nl.utwente.viskell.haskell.expr.Binder;
+import nl.utwente.viskell.haskell.expr.LetExpression;
+
+// TODO make this an independent class if other blocks (case?) need this too 
+/** An internal output anchor for an argument binder. */
+public class BinderAnchor extends OutputAnchor {
+
+    /**
+     *
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/BinderAnchor.java]: Empty comment.

File: Code/src/main/java/nl/utwente/viskell/ui/components/BinderAnchor.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    @Override
+    public void prepareConnectionChanges() {
+        this.lambdaContainer.refreshAnchorTypes();
+    }
+
+    @Override
+    protected void handleConnectionChanges(boolean finalPhase) {
+        this.lambdaContainer.handleConnectionChanges(finalPhase);
+    }
+}
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/BinderAnchor.java]: Missing newline end of file (also for other files).

File: Code/src/main/java/nl/utwente/viskell/ui/components/BlockContainer.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package nl.utwente.viskell.ui.components;
+
+public interface BlockContainer {
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/BlockContainer.java]: Missing comment (and the interface doesn't really seem to be used?)

File: Code/src/main/java/nl/utwente/viskell/ui/components/LambdaContainer.java [hunk 4]
```diff
@@ -145,18 +76,18 @@ public LambdaContainer(DefinitionBlock wrapper, String name, Type type) {
...
+        this.res = new ResultAnchor(this, wrapper, Optional.of(t));
 
         this.argSpace.getChildren().addAll(this.args);
         this.resSpace.getChildren().add(this.res);
     }
 
-    /** Set fresh types in all anchors of this lambda for the next typechecking cycle. */
-    protected void refreshAnchorTypes() {
+    @Override
+    public void refreshAnchorTypes() {
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/LambdaContainer.java]: Does this have to be `public`?

File: Code/src/main/java/nl/utwente/viskell/ui/components/BinderAnchor.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+// TODO make this an independent class if other blocks (case?) need this too 
+/** An internal output anchor for an argument binder. */
+public class BinderAnchor extends OutputAnchor {
+
+    /**
+     * 
+     */
+    private final LambdaContainer lambdaContainer;
+
+    // FIXME BinderAnchor should not have or use the DefinitionBlock parent
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/viskell/ui/components/BinderAnchor.java]: FIXME seems to be outdated (same for duplicated comment in `ResultAnchor` class.)

**Comment by wandernauta** [comment]
I don't see any major problems with this. Will merge if you fix the (small) issues above.

File: Code/src/main/java/nl/utwente/viskell/ui/components/LambdaContainer.java (hunk 4 above)
Comment (Flumble) [inline Code/src/main/java/nl/utwente/viskell/ui/components/LambdaContainer.java]: Unfortunately, you can't have protected methods (or attributes) in an interface. And making it an abstract class is not an option either.

**Pull request merged**